### ISSUE TITLE
Explain how (not) to upgrade OpenSaas

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/open-saas#240]
We had a couple of questions how to best upgrade existing OpenSaas project when new version comes out. We should make it clear you are not really meant to easily upgrade it, but that you might want to cherry-pick changes that you like, as it is a starter project.

**Comment by marcoeg** [comment]
Thanks for the clarification. Please recommend the best course of action in creating a new project. Is the `wasp new` and then selecting `saas` generating the same repo as forking the latest `open-saas` repo straight on? Let me take the opportunity to also let you know that the difference between the saas template, the saas demo code and the saas template diff, is quite confusing. In any case, this is a great project you are doing in addition to Wasp and it deserves full credit. I am in the process to start at least two projects based on it. Happy to support the initiative in any way I can, including subscribing to your paid services.

**Comment by Martinsos** [comment]
`wasp new -t saas` is the way to go! Not cloning the repo directly.

template vs demo code vs docs -> yup that is confusing indeed, atlhgouh it is aimed for us core developers, you as an opensaas user shouldn't have to worry about that! We are not yet sure if we will stick with the app_diff approach, it is a bit uncommon approach, something we are experimenting with.
How did you end up looking into demo app / diffs?

**Comment by marcoeg** [comment]
Got it. Thank you for the reply. I have been looking at the repo because of the relatively high frequency of commits. An indication that there is still work in progress. Generating a new saas project from the CLI, what is the suggested way to merge the new commits into the project?

**Comment by Martinsos** [comment]
Thanks for letting me know! Ah yes we are always improving something :). That doesn't neccesarily mean that current version is in bad shape, just that we are getting new ideas how to do stuff better. We are also keeping it up to date with new versions of Wasp.

Merging new commits -> we don't recommend doing it, opensaas is not meant to be updated, it is a starter. If you really do want to pull in the changes, it is best to do it manually -> take a look at the new code (either in template/ or by creating new project on your machine) and pick what you like. Or alternatively go the opposite direction: create new fresh project from CLI and carry over your own code from older project, whatever works easier.

**Comment by marcoeg** [comment]
Okay. May I suggest to adopt at some point the idea of releases to facilitate such process? If nothing else, having release notes and release dates helps in understanding whether updating may be helping an existing application or not and the cherry picking process you are referring to. Cheers.

**Comment by Martinsos** [comment]
@marcoeg yes that will probably make sense in the future! We still haven't figured out exactly how we want to do the releases, but we will probably do it. Here is na issue for it, and I linked to your comment: https://github.com/wasp-lang/open-saas/issues/62 . Thanks!

**Comment by Martinsos** [comment]
@genyus made a nice guide for this: https://discord.com/channels/686873244791210014/1355980920125001898/1356001024690618575 .
